### ISSUE TITLE
Test with the new drop-down for the Wayland/X11 question

### ISSUE THREAD

**Issue by Elliria** [Elliria/autokey-fork#8]
### Do you use X11 or Wayland?

Wayland

### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

None

### Which AutoKey GUI did you use?

None

### Can you briefly describe the issue?

This example demonstrates the Wayland/X11 question presented as a drop-down, limiting the user to only one choice.

If I make the question not required, GitHub adds the "None" choice to the drop-down and makes it the default choice if no selection is made by the user.

If I make the question required, then GitHub doesn't add the "None" choice to the drop-down and, instead, makes the first item listed the default choice. This would be risky regardless of which of the two choices we chose, so I added the "Unknown" entry to the list so that GitHub will make that the default answer if no selection is made by the user.

I also got rid of the second red emoji, because with all that red going on, it was busy enough that the user could overlook the red asterisk provided by GitHub for required items.

You'll want to start an issue to see if you like it.

It also occurs to me that we could put your statement at the beginning of the question, so that instead of saying:

**Do you use X11 or Wayland?**

The question could be:

**AutoKey is an X11 application and will not function on distributions that default to using Wayland instead of Xorg. Do you use X11 or Wayland?**

That might neatly solve the whole thing, because it would stand out nicely.

### Can the issue be reproduced?

None

**Comment by josephj11** [comment]
Just looked at a new issue and it's as good as it's going to get. Thanks for all your work.

One thing that bugs me is the wording. I got it from somewhere and you inherited it.

It would be better if it said "configured for" instead of "default to".  The default doesn't matter. What matters is which one is active when AutoKey runs.

**Comment by Elliria** [comment]
I simplified the statement some more and made it part of the first question on the form.

By the way, something odd that I noticed, that's true on any GitHub YAML issue template,  is that neither questions with drop-downs nor questions with multiple check-boxes can be made required.

If you put the **required: true** attribute into the **validations** field in the drop-down, GitHub automatically marks that question with a red asterisk to indicate to the user that an answer is required, but the form can be submitted without answering the question.

If you put the **required: true** attribute into the **validations** field in the in a question with multiple check-boxes, GitHub doesn't automatically mark the question with a red asterisk, which indicates to the user that answering the question optional.

With that in mind, the drop-down is set as required, but the user can ignore that [sigh].

**Comment by Elliria** [comment]
By the way, I filed [this bug report](https://github.com/community/community/discussions/45084) about that issue. Maybe they'll fix it.

**Comment by josephj11** [comment]
Gave the new issue an upvote.

**Comment by Elliria** [comment]
I agree, and it was also surprising that I was the first person to report it. You'd think that many of the thousands (or more?) of GitHub users would have noticed it, too, especially since this new format for issue reports is so exciting in comparison to what we had before.

**Comment by Elliria** [comment]
I'm glad you like the latest changes. Do you think we should also get rid of the emoji at the beginning of the Xorg/Wayland question or should we leave it there?

By the way, I've done so many experiments with this that I'm thinking I'll get rid of this fork entirely and do a fresh fork so I can do a pull request with just one commit added to it instead of a history of them.

**Comment by josephj11** [comment]
I think there's a git command, something like squash that compresses all commits into one.

If we're looking at the same thing, it looks like an octagon with something inside it that's too small to see. I'm fine with it being there, but it's not important to me.

**Comment by Elliria** [comment]
Hey there,

Joe wrote:

>I think there's a git command, something like squash that compresses
>all commits into one.

Yeah, BlueDrink did it at that end the last time around. Do you know
how it's done? If so, I'll submit a pull request as it sits.
Otherwise, I'm not confident in my ability to do it from this end, so
I'd probably just destroy this fork and create a new one with one
clean commit in it and submit a pull-request from there so that
neither of us has to do any clean-up.

>If we're looking at the same thing, it looks like an octagon with
>something inside it that's too small to see. I'm fine with it being
>there, but it's not important to me.

Okay, then I'll remove it so that all the questions look alike aside
from GitHub's automatically-added asterisks for required items.

-- 
Little Girl

There is no spoon.

**Comment by josephj11** [comment]
I know nothing, but this is what the  internet says

https://www.git-tower.com/learn/git/faq/git-squash

https://www.baeldung.com/ops/git-squash-commits

https://www.jenweber.dev/how-to-squash-and-rebase/

Once you have a copy of your current branch saved aside, you have nothing to lose by trying one of the above as a learning experiment.

**Comment by Elliria** [comment]
Thanks. Here's what I've got after some research:

According to [About pull request merges](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges), "To squash and merge pull requests, you must have write permissions in the repository, and the repository must allow squash merging." Apparently, the AutoKey repository allows squash merging already, but I don't have write permissions in the repository.  When I submitted my previous pull request, BlueDrink turned it into a squash merge at that end.

[This page](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/merging-a-pull-request#merging-a-pull-request) has the instructions for what needs to happen at your end. I summed them up here, too:

1. Under your repository name, click "Pull requests".
2. In the "Pull Requests" list, click the pull request you'd like to merge.
3. Choose the "Squash and merge" option from the drop-down menu.
4. Click the "Squash and merge" button.
5. If prompted, type a commit message or accept the default message.
6. If you have more than one email address associated with your GitHub account, click the email address drop-down menu and select the email address to use as the Git author email address.
7. Click "Confirm squash and merge".
8. Optionally, [delete the branch](https://docs.github.com/en/articles/deleting-unused-branches).

I'm game to try it if you like. Shall we go for it?

**Comment by josephj11** [comment]
From what I read, I was under the impression that you could do the squash with the rebase command on your end. If that won't work, then we can do the merge squash and let me be the one to break everything. LOL.

**Comment by Elliria** [comment]
I'm not well-versed in all of this, but from what I've been able to learn, rebasing isn't what I'm after and the software that the instructions are often for is for Windows or Mac, neither of which I'm using. It would definitely want to be squashed.

I get the feeling we're both worried about breaking things and not necessarily having a handy rescuer to get us out of it. If you don't mind, I've made a local copy of the ```bug.yaml``` file and can just delete this fork entirely and do a fresh fork. The only thing I've ever done with this fork was write the ```bug.yaml``` file and it's already been merged into the actual AutoKey code, so my fork is unnecessary and BlueDrink had told me back then that I could delete it. I just didn't get around to it, so it's still here. As long as you don't mind losing our conversations back and forth in here, we could do a safe single-commit that wouldn't need to be squashed with a fresh fork.

Also, in the future, if we need to do something like this again, I'll do this, which should be air-tight:
1. Fork the remote repository.
2. Create a new local repository.
3. Copy a file from the fork into the local repository.
4. Work on the local repository's file until it's solid.
5. Copy the final code from the local repository's file into the fork's file.
6. Commit it.
7. Do a pull-request from the fork to the remote repository.
8. Wait for it to be accepted at the remote repository's end.
9. Delete the fork.
10. Delete the local repository.

**Comment by josephj11** [comment]
OK. There's nothing precious in our comments here, so whenever you're ready you can use the method above starting over and replacing the original form with a copy saved from here. Then I should be able to merge it easily. I did one or two of them before Bluedrink9 asked for and got full permissions.

**Comment by Elliria** [comment]
Okay. I'm going to try it.
